space-concordia-robotics/robotics-prototype
# Science design 462

# Assignee Section

## Description
Design and skeleton of teensy science code.

- Design documents in PlantUML code in `docs/`
- Added Skeleton code for C++

Notes : 

1. I added some code to the CommandCenter but it's not within the scope of this PR to make it work. I would also like to test it with unit tests.
2. There's some OOP and a malloc in this c++, in both cases I think the memory usage will be kept low

### Steps for Testing
1. Copy paste the diagram code (science-class.uml and science-deployment.uml) in here to see a pic http://www.plantuml.com/plantuml/uml/SyfFKj2rKt3CoKnELR1Io4ZDoSa70000
2. Run the science code and make sure it compiles https://github.com/space-concordia-robotics/robotics-prototype/wiki/Building-the-C---code

closes #462 

The approval from all software team leads is necessary before merging.

# Reviewer Section

Aside from local testing and the General Integration Test it is implied that static analysis should be included in the verification process.


For Pull Requests that do not include code changes, it is not required to perform the tests above.
